java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java

```diff
@@ -0,0 +1,403 @@
+   * In such cases, we are likely to face OOM or related problems when
+   * we reserve memory for a record batch with value count x and
+   * do setInitialCapacity(x) such that each vector allocates only
+   * what is necessary and not the default amount, but the multiplier
+   * forces the memory requirement to go beyond what was needed.
+   *
+   * @param numRecords value count
+   * @param totalNumberOfElements the total number of elements to allow
+   *                              for in this vector across all records.
+   */
+  public void setInitialTotalCapacity(int numRecords, int totalNumberOfElements) {
+    offsetAllocationSizeInBytes = numRecords * OFFSET_WIDTH;
+    sizeAllocationSizeInBytes = numRecords * SIZE_WIDTH;
+    vector.setInitialCapacity(totalNumberOfElements);
+  }
+
+  @Override
+  public int getValueCapacity() {
+    return 0;
+  }
+
+  protected int getOffsetBufferValueCapacity() {
+    return capAtMaxInt(offsetBuffer.capacity() / OFFSET_WIDTH);
+  }
+
+  protected int getSizeBufferValueCapacity() {
+    return capAtMaxInt(sizeBuffer.capacity() / SIZE_WIDTH);
+  }
+
+  @Override
+  public int getBufferSize() {
+    if (valueCount == 0) {
+      return 0;
+    }
+    return (valueCount * OFFSET_WIDTH) + (valueCount * SIZE_WIDTH) + vector.getBufferSize();
+  }
+
+  @Override
+  public int getBufferSizeFor(int valueCount) {
+    if (valueCount == 0) {
+      return 0;
+    }
+
+    int innerVectorValueCount = 0;
+
+    for (int i = 0; i < valueCount; i++) {
+      innerVectorValueCount += sizeBuffer.getInt(i * SIZE_WIDTH);
+    }
+
+    return (valueCount * OFFSET_WIDTH) + (valueCount * SIZE_WIDTH) +
+        vector.getBufferSizeFor(innerVectorValueCount);
+  }
+
+  @Override
+  public Iterator<ValueVector> iterator() {
+    return Collections.<ValueVector>singleton(getDataVector()).iterator();
+  }
+
+  @Override
+  public void clear() {
+    offsetBuffer = releaseBuffer(offsetBuffer);
+    sizeBuffer = releaseBuffer(sizeBuffer);
+    vector.clear();
+    valueCount = 0;
+    super.clear();
+  }
+
+  @Override
+  public void reset() {
+    offsetBuffer.setZero(0, offsetBuffer.capacity());
+    sizeBuffer.setZero(0, sizeBuffer.capacity());
+    vector.reset();
+    valueCount = 0;
+  }
+
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    return new ArrowBuf[0];
+  }
+
+  @Override
+  public int getValueCount() {
+    return valueCount;
+  }
+
+  @Override
+  public boolean isNull(int index) {
+    return false;
+  }
+
+  @Override
+  public void setValueCount(int valueCount) {
+    this.valueCount = valueCount;
+    while (valueCount > getOffsetBufferValueCapacity()) {
+      reallocateBuffers();
+    }
+    final int childValueCount = valueCount == 0 ? 0 : getLengthOfChildVector();
+    vector.setValueCount(childValueCount);
+  }
+
+  private int getLengthOfChildVector() {
+    int length = 0;
+    for (int i = 0; i <= valueCount; i++) {
+      length += sizeBuffer.getInt(i * SIZE_WIDTH);
```

Comment:
This is wrong. Elements can overlap. I think you need to compute length as `max(offset + length) - min(offset)`

Reply:
what if we just use `vector.getValueCount()`, this is the childVector or `vector` as named in the `BaseRepeatedValueVector`

Reply:
If this is correct (_I think so_), we can add validations to `setOffSet` and `setSize` quite easily.

Reply:
It depends on how this is used. Are you using this as the physical length, or do you only care about the part that the list vector actually references?

Reply:
`max(offset + length) - min(offset)` or `max(offset + size) - min(offset)`

Reply:
@lidavidm is this right?